rotors_gazebo_plugins/src/gazebo_noisydepth_plugin.cpp

```diff
@@ -0,0 +1,308 @@
+// Increment count
+void GazeboNoisyDepth::DepthImageConnect() {
+  this->depth_image_connect_count_++;
+  this->parentSensor->SetActive(true);
+}
+////////////////////////////////////////////////////////////////////////////////
+// Decrement count
+void GazeboNoisyDepth::DepthImageDisconnect() {
+  this->depth_image_connect_count_--;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Increment count
+void GazeboNoisyDepth::DepthInfoConnect() {
+  this->depth_info_connect_count_++;
+}
+////////////////////////////////////////////////////////////////////////////////
+// Decrement count
+void GazeboNoisyDepth::DepthInfoDisconnect() {
+  this->depth_info_connect_count_--;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Update the controller
+void GazeboNoisyDepth::OnNewDepthFrame(const float *_image,
+                                       unsigned int _width, unsigned int _height, unsigned int _depth,
+                                       const std::string &_format) {
+  if (!this->initialized_ || this->height_ <= 0 || this->width_ <= 0)
+    return;
+
+  this->depth_sensor_update_time_ = this->parentSensor->LastMeasurementTime();
+  if (this->parentSensor->IsActive()) {
+    if (this->depth_image_connect_count_ <= 0 &&
+        (*this->image_connect_count_) <= 0) {
+      this->parentSensor->SetActive(false);
+    } else {
+      if (this->depth_image_connect_count_ > 0)
+        this->FillDepthImage(_image);
+    }
+  } else {
+    if (this->depth_image_connect_count_ <= 0)
+      // do this first so there's chance for sensor to run 1 frame after activate
+      this->parentSensor->SetActive(true);
+  }
+  PublishCameraInfo();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Update the controller
+void GazeboNoisyDepth::OnNewImageFrame(const unsigned char *_image,
+                                       unsigned int _width, unsigned int _height, unsigned int _depth,
+                                       const std::string &_format) {
+  if (!this->initialized_ || this->height_ <= 0 || this->width_ <= 0)
+    return;
+
+  //ROS_ERROR_NAMED("openni_kinect", "camera_ new frame %s %s",this->parentSensor_->Name().c_str(),this->frame_name_.c_str());
+  this->sensor_update_time_ = this->parentSensor_->LastMeasurementTime();
+
+  if (this->parentSensor->IsActive()) {
+    if (this->depth_image_connect_count_ <= 0 &&
+        (*this->image_connect_count_) <= 0) {
+      this->parentSensor->SetActive(false);
+    } else {
+      if ((*this->image_connect_count_) > 0)
+        this->PutCameraData(_image);
+    }
+  } else {
+    if ((*this->image_connect_count_) > 0)
+      // do this first so there's chance for sensor to run 1 frame after activate
+      this->parentSensor->SetActive(true);
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Put depth image data to the interface
```

Comment:
This belongs rather to the header file :) (and is probably not adding so much information)